core/src/main/java/nordpol/Apdu.java

```diff
@@ -108,7 +107,8 @@ public static String encodeHex(byte b) {
      * @return true if the APDUs match, false if not
      */
     public static boolean hasStatus(byte[] receivedApdu, byte[] statusCode) throws IOException {
-        return hasStatus(encodeHex(receivedApdu), encodeHex(statusCode));
+        byte[] receivedStatus = statusBytes(receivedApdu);
+        return receivedStatus[0] == statusCode[0] && receivedStatus[0] == statusCode[0];
```

Comment:
@Franzaine Thanks, a bit of an epic fail one could say... It's also clear that the unit tests are lacking not catching this :-)